### PR TITLE
Upgrade PHPStan and extend PrestaShop rules 

### PR DESCRIPTION
**Pull request by Quetzacoalt91**
This PR upgrades PHPStan to v0.12 which includes more rules and provide more feedback, thus the CI broken.

It also updates the way we configure PrestaShop modules specific rules. We now extend a configuration file from `prestashop/php-dev-tools` library, which will define additional rules and fixes ready to be deployed on all modules at the same time.

### CODE REVIEW

File: controllers/front/DispatchWebHook.php [hunk 1]
```diff
@@ -36,9 +36,9 @@ class ps_checkoutDispatchWebHookModuleFrontController extends ModuleFrontControl
     public $auth = false;
 
     /**
-     * Id coming from PSL
+     * Id coming from PSL (CheckMe: Is it a UUID?)
```
Comment (Quetzacoalt91) [inline controllers/front/DispatchWebHook.php:39]: @Matt75 / @v4lux, can you please confirm this property stores a string?

Comment (Matt75) [reply controllers/front/DispatchWebHook.php:39]: Yes, it's a string

**Comment by Matt75** [comment]
@Quetzacoalt91 This PR should be rebased

**Comment by Quetzacoalt91** [comment]
Here you go!

**Comment by Matt75** [comment]
Thanks you @Quetzacoalt91

**Pull request merged**